### PR TITLE
ci: add test to make sure env vars in .env.example.holesky are valid

### PR DESCRIPTION
**Pull request by samlaf**
## Fixes Issue

Fixes #

## Changes proposed

Mantle has run into issues because one of the flags in our .env.holesky.example file was incorrect.
> In the configuration file example and the readme file, the variable name has always been EIGENDA_PROXY_EIGENDA_ETH_CONFIRMATION_DEPTH.
> https://github.com/Layr-Labs/eigenda-proxy/blob/97b71bda6e2136a6c0d71be2f7e9642dc840fefc/.env.example.holesky
> https://github.com/Layr-Labs/eigenda-proxy/blob/main/README.md

This PR removes the flags in the README which are just very prone to be outdated and induce errors like this, and also adds a ci check to make sure the flags defined in .env.example.holesky are valid. This test is simple and will only catch if that file still contains deprecated flags (which cause the proxy to crash on startup); it won't be able to catch the subtle case above where the flag is not an old deprecated flag but instead just an unused env var.

### CODE REVIEW

File: scripts/test-proxy-startup-with-env-vars.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -e  # Exit on any error
+
+##### This script is meant to be run in ci #####
+# It tests that the env vars defined in .env.example.holesky are correct.
+# It starts the eigenda-proxy with those env vars, waits 5 seconds, and then kills the proxy.
+# If any deprecated flags are still being used in .env.example.holesky, the script will fail.
```
Comment (ethenotethan) [inline scripts/test-proxy-startup-with-env-vars.sh:7]: shouldn't we also test set/get logic as well? or at a minimum ping the health endpoint to at least ensure the server instance successfully finished bootstrapping?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -17,8 +17,23 @@ jobs:
       with:
         go-version: 1.22
 
-    - name: Build App
+    - name: Check that build succeeds
       run: make eigenda-proxy
+  
+  flags:
```
Comment (ethenotethan) [inline .github/workflows/build.yml:23]: knit: `holesky-flags`

**Review by bxue-l2 — DISMISSED** [review]

File: scripts/test-proxy-startup-with-env-vars.sh (hunk 1 above)
Comment (samlaf) [reply scripts/test-proxy-startup-with-env-vars.sh:7]: [9204b0d](https://github.com/Layr-Labs/eigenda-proxy/pull/286/commits/9204b0d5d7dc2ae8e62d8cbb4b73c083423ed3c2)

**Review by ethenotethan — APPROVED** [review]
LGTM
